apps/frontend/src/components/TimeCardPage/CellTypes/CommentModals/ShowCommentModal.tsx

```diff
@@ -174,8 +201,18 @@ export default function ShowCommentModal({
       };
   
       const handleSubmit = () => {
-        // TODO: reuse comment validation
-        setComments([...comments, createNewComment(user, CommentType.Comment, remark)]);
+        
+        var rowToMutate = row.Comment; 
+        if (rowToMutate === undefined) {
+            rowToMutate = comments;
+        }
+        // Value is null, so mark it as undefined in our processing 
+        rowToMutate = setComments([...comments, createNewComment(user, CommentType.Comment, remark)]);
+        console.log(rowToMutate)
```

Comment:
Make sure to remove any extra `console.log()` before you merge in

Reply:
e1f9579a42a756a9d442c7797c9d3b9ef29e7537